src/deroff.rs

```diff
@@ -376,18 +376,12 @@ impl Deroffer {
         self.s = self.s.trim_start().to_owned();
     }
 
-    fn str_at(string: &str, idx: usize) -> &str {
-        // Note: If we don't care about strings with multi-byte chars, the
-        // following would suffice:
-        // s.get(idx..idx + 1).unwrap_or("")
-        //
-        // Note: We're not yet sure whether our roff inputs will generally be
-        // ASCII or UTF-8. If they are ASCII (and can be treated as containing
-        // only single-byte characters), it would be faster to just use `get()`
-        string
-            .char_indices()
-            .nth(idx)
-            .map(|(idx, charr)| &string[idx..(idx + charr.len_utf8())]) // Okay to directly index based on idx/charr construction.
+    fn str_at(&self, idx: usize) -> &str {
+        let s = &self.s;
+        s.char_indices()
+            .skip(idx)
+            .next()
+            .map(|(i, c)| &s[i..(i + c.len_utf8())])
```

Comment:
Do we even need to handle all UTF? I thought utf8 ascii position are still in place? I believe we are only checking those markups which is at most one usize.

Reply:
This was a @scooter-dangle thing. I just reformatted it to work in #75